packages/jsii-reflect/lib/type-system.ts

```diff
@@ -316,6 +316,18 @@ export class TypeSystem {
     return new Assembly(this, ass);
   }
 
+  private getAssemblyFile(pathToAssembly: string) {
+    const jsiiFile = path.join(pathToAssembly, '.jsii');
+    const compressedJsiiFile = path.join(pathToAssembly, '.jsii.gz');
+    const jsiiExists = fs.existsSync(jsiiFile);
+    if (!jsiiExists && !fs.existsSync(compressedJsiiFile)) {
```

Comment:
If they both exist... there might be an issue -- in particular, this will ALWAYS prefer loading the uncompressed assembly, when it should likely prefer loading the compressed one if it exists...

A possible solution would be to *ALWAYS* load `.jsii`, but if there is a compressed assembly, that file basically contains something along the lines of:

```json
{
  "schema": "jsii/file-redirect",
  "compression": "gzip",
  "filename": ".jsii.gz"
}
```

The loader would detect this... and proceed to load (and uncompress) `.jsii.gz` as indicated.

Reply:
new PR for `@jsii/utils` does do something like this.